### ISSUE TITLE
Any Training  instructions?

### ISSUE THREAD

**Issue by albertyou2** [jwyang/JULE-Torch#6]
HI @jwyang 
Great work,and I think this is what I am  looking for  !
But I found that it is a little difficult for me to train custom dataset .Could your please give some help?

My question is :
How to create training dataset?What 's the format of the datas?
Now i want train my custom images,I have no idea how to  transform them to .h5 file
could you please give a step by step training instructions()?I think this will help other people too.
Thank you very much!!!!

**Comment by jwyang** [comment]
Hi, @albertyou2 

Thanks, regarding to your questions:

 How to create training dataset?What 's the format of the datas?
-> the dataset should contain train images and train labels. Mostly, I used matlab to read images and then combine them into a single tensor, whose dimension is Height x Width x Channel x Nsamples.  Along with train data, the labels should have dimension 1 x Nsamples. Generally, we need to whiten the data. But it is not always better than doing nothing from my experiments. After that, you can save them to a hdf5 file with matlab.

 Now i want train my custom images,I have no idea how to transform them to .h5 file
-> As illustrated above, you need to organize the images to a single tensor, and then combine their cluster labels to create a hdf5 file. Then create your own folder in the datasets folder with name of your dataset, and name the hdf5 file to 'data4torch.h5'. Then you are all set to run the code.

 could you please give a step by step training instructions()?I think this will help other people too.
-> Yes, I will elaborate how to train on your own dataset. Thanks for reminding.

**Comment by albertyou2** [comment]
@jwyang Thanks for your reply !
I will have a try .

**Comment by bnascimento** [comment]
Since this is unsupervised clustering, why do we need the labels? why not just the images? For testing purposes?
How can I setup a dataset with just images and no labels? is that possible? cheers and thank you

**Comment by jwyang** [comment]
Yes, the labels are used for evaluation only. To cluster images without labels, just set the label randomly and comment evalPerf() if you want.

**Comment by bnascimento** [comment]
Would you be able to share some code how to transform the data from images to NxCxWxH tensor and into hdf5 dataset?

**Comment by jwyang** [comment]
Thanks for the suggestions, I will do that, and push the code to this repository soon.

**Comment by ShansYu** [comment]
Hi~@jwyang I have the same problem. How to transform the data from images to NxCxWxH tensor and into hdf5 dataset? I can't find the code as you said above.

**Comment by jwyang** [comment]
Hi, @ShansYu, yes, you are right, I have not yet uploaded the code for transformation. My previous code is based on matlab, and missed after I reinstall my system. It should be very simple to load images from folder based on torch. Please refer to this [demo code](https://github.com/torch/demos/blob/master/load-data/load-images.lua). You can create a tensor to hold all images data, and another tensor to hold labels you have the labels.

**Comment by ShansYu** [comment]
@jwyang Thanks for your reply ! I will have a try.

**Comment by jwyang** [comment]
@ShansYu  sounds good!
